scripts/run-framework/agief_experiment/experiment.py

```diff
@@ -467,25 +468,25 @@ def upload_results(self, cloud, compute_node, export_compute):
                 logging.warning("No data file found. This should only happen if you are running remote via ssh, " \
                       "and exporting data by saving on compute.")
             else:
-                # Get features and labels CSV files
-                output_labels_filepath = utils.match_file_by_name(folder_path, self.LABELS_FILENAME)
-                output_features_filepath = utils.match_file_by_name(folder_path, self.FEATURES_FILENAME)
+                files_to_compress = [output_data_filepath]
+                archive_filename = self.experiment_utils.outputfile(self.prefix(), "data.zip")
 
-                files_to_compress = [
-                    output_data_filepath,
-                    output_labels_filepath,
-                    output_features_filepath
-                ]
+                if self.csv_output:
+                    # Get features and labels CSV files
+                    output_labels_filepath = utils.match_file_by_name(folder_path, self.LABELS_FILENAME)
+                    output_features_filepath = utils.match_file_by_name(folder_path, self.FEATURES_FILENAME)
 
-                archive_filename = self.experiment_utils.outputfile(self.prefix(), "data.zip")
+                    files_to_compress.append(output_labels_filepath)
+                    files_to_compress.append(output_features_filepath)
 
-                # Compress data, features and labels
+                # Compress the output files
                 utils.compress_files(archive_filename, files_to_compress)
```

Comment:
what if `files_to_compress` is empty?

Reply:
It wouldn't be in this case, since it will have `files_to_compress = [output_data_filepath]` and we've already checked for the existence of that file